### PR TITLE
[PORT] Lazy Template loading for antagonist structures

### PR DESCRIPTION
**Pull request by PestoVerde322**
## About The Pull Request

*This pr is a port of:

- https://github.com/tgstation/tgstation/pull/71785
- https://github.com/tgstation/tgstation/pull/72339
- https://github.com/tgstation/tgstation/pull/59328
- https://github.com/tgstation/tgstation/pull/72800

This PR removes the antagonists spawn areas from the Centcomm map, and instead they are lazy loaded.

To make a new lazy load template is fairly simple, make a map, allocate an area for it, and place a marker at the bottom left corner of that area. I have it check an area to ensure that if someone makes the map larger than expected but doesn't account for the template allocation it doesn't overwrite stuff without warning.

Also it replaces some improper CHECK_TICKs with MAPLOADING_CHECK_TICKs and also 
delete some code that solved an issue that was present ages ago, but nowadays it's no longer the case.

## Why It's Good For The Game

Trims down on int times overall.

## Testing Photographs and Procedure

N/A, what am I supposed to show, a more empty Centcomm?

## Changelog
:cl:
code: Centcomm map has been cleared out, now all the antitag structure will load whenever certain antitags spawns (example nuke ops)
admin: New mapping verb to load lazy templates as needed. 
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Show the maps loading properly when they are meant to

File: code/datums/lazy_template.dm
Comment (PowerfulBacon) [inline code/datums/lazy_template.dm]: This is not going to work due to our changes to map template loading. Our map template loading doesn't sleep, it instead returns a datum which represents the future result of the map loading. This means that if you want to use the results of a .load, you need to use on_completion inside of the returned datum or block until its completed.
Our map loading is lazy by default.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/datums/lazy_template.dm
Comment (PowerfulBacon) [reply code/datums/lazy_template.dm]: We should probably refactor the return value to be part of the async framework which would clean things up a bit. If that's needed then I can work on that since it would enable just adding AWAIT(loading.load(reservation_bottom_left)) instead of currently which is to either use callbacks or UNTIL

**Comment by PowerfulBacon** [comment]
This will need proper testing to ensure that
A) The maps load properly when requested
B) The procs block as expected
C) Things like the syndicate and abductors actually get teleported to the loaded templates

Our maploading is quite different from TG's as it does not use CHECK_TICK but instead runs on a subsystem in the background as an async task.

**Review by PowerfulBacon — APPROVED** [review]
This is probably fine

**Comment by PestoVerde322** [comment]
> This is probably fine

I still would like to issue a TM before merging to see if it's fully stable.

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm
Comment (itsmeow) [inline code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm]: This is already handled by `..()`.

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm
Comment (itsmeow) [inline code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm]: this is also already handled by `ready()`. acceptable should not fail for invalid candidates

**Review by itsmeow — CHANGES_REQUESTED** [review]
This should also cover Reebe, by far the laggiest load.

I would like this testmerged.

**Comment by itsmeow** [comment]
Can you show it working locally with a forced dynamic ruleset?

**Comment by PowerfulBacon** [comment]
> This should also cover Reebe, by far the laggiest load.

Reebe is already 'lazyloaded' by the definition in this PR. This PR isn't adding asynchronous loading, its adding the ability to load maps which aren't spawned roundstart which Reebe already does. Making Reebe work better would require another PR porting over the logic in the normal clockcult gamemode which spawns the map and picks the mode 30 seconds before roundstart to prevent roundstart lag.

**Comment by itsmeow** [comment]
Address my dynamic requests and we can testmerge this

File: code/game/gamemodes/nuclear/nuclear.dm
Comment (itsmeow) [inline code/game/gamemodes/nuclear/nuclear.dm]: ~~Can we move this? It's silly. Put this in the actual gamemode instead, not the goddamn outfit equipping.~~

~~Also, the wizard gamemode (does that exist?) or round event? needs a trigger. Same for nukeops round event~~

Yeah that's already done, just delete this please.

**Comment by itsmeow** [comment]
You appear to have removed the cargo shuttle bay?
image

**Comment by PestoVerde322** [comment]
> You appear to have removed the cargo shuttle bay? image

immagine

Nope, it's now attached to the Centcom dock!

**Comment by PestoVerde322** [comment]
On hold untill all the genpop issues are addressed.

**Comment by Rukofamicom** [comment]
are genpop issues addressed yet? It's been quite a while

**Pull request closed without merging**
